src/simplewallet/simplewallet.cpp

```diff
@@ -2156,7 +2156,65 @@ bool simple_wallet::transfer_main(int transfer_type, const std::vector<std::stri
           prompt << boost::format(tr(".\nThis transaction will unlock on block %llu, in approximately %s days (assuming 2 minutes per block)")) % ((unsigned long long)unlock_block) % days;
         }
-        prompt << tr(".") << ENDL << tr("Is this okay?  (Y/Yes/N/No): ");
+        std::string err;
+        uint64_t blockchain_height = get_daemon_blockchain_height(err);
+        if (!err.empty())
+        {
+          fail_msg_writer() << tr("failed to get blockchain height: ") << err;
+          return true;
+        }
+        // for each transaction
+        prompt << tr("\nNumber of transactions: ") << ptx_vector.size();
+        for (size_t n = 0; n < ptx_vector.size(); ++n)
+        {
+          const cryptonote::transaction& tx = ptx_vector[n].tx;
+          const tools::wallet2::tx_construction_data& construction_data = ptx_vector[n].construction_data;
+          prompt << boost::format(tr("\nTransaction #%llu: txid=%s")) % n % cryptonote::get_transaction_hash(tx);
+          // for each input
+          prompt << tr("\nNumber of inputs: ") << tx.vin.size();
+          for (size_t i = 0; i < tx.vin.size(); ++i)
+          {
+            if (tx.vin[i].type() != typeid(cryptonote::txin_to_key))
+              continue;
+            const cryptonote::txin_to_key& in_key = boost::get<cryptonote::txin_to_key>(tx.vin[i]);
+            const cryptonote::tx_source_entry& source = construction_data.sources[i];
+            prompt << boost::format(tr("\nInput #%llu: amount=%s")) % i % print_money(source.amount);
+            // convert relative offsets of mixin keys into absolute offsets (indices) associated with the amount
+            std::vector<uint64_t> absolute_offsets = cryptonote::relative_output_offsets_to_absolute(in_key.key_offsets);
+            // get block heights from which those keys in the ring originated
+            COMMAND_RPC_GET_OUTPUTS_BIN::request req = AUTO_VAL_INIT(req);
+            req.outputs.resize(absolute_offsets.size());
+            for (size_t i = 0; i < absolute_offsets.size(); ++i)
+            {
+              req.outputs[i].amount = in_key.amount;
+              req.outputs[i].index = absolute_offsets[i];
+            }
+            COMMAND_RPC_GET_OUTPUTS_BIN::response res = AUTO_VAL_INIT(res);
+            bool r = net_utils::invoke_http_bin_remote_command2(m_wallet->get_daemon_address() + "/get_outs.bin", req, res, m_http_client);
+            err = interpret_rpc_response(r, res.status);
+            if (!err.empty())
+            {
+              fail_msg_writer() << tr("failed to get output: ") << err;
+              return true;
+            }
+            prompt << tr("\nOriginating block heights: ");
+            for (size_t i = 0; i < absolute_offsets.size(); ++i)
+              prompt << tr(i == source.real_output ? " *" : " ") << res.outs[i].height;
+            // visualize the distribution, using the code by moneroexamples onion-monero-viewer
+            const uint64_t resolution = 100;
```

Comment:
It was 170 in the original code of onion-monero-viewer and I reduced it to 100 so that it fits in my display:)
Why do you bring up a particular number 80?